src/main/java/com/lambdatest/jenkins/freestyle/report/AppAutomationReportBuildAction.java

```diff
@@ -0,0 +1,135 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import hudson.model.Run;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.lambdatest.jenkins.freestyle.api.Constant;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.*;
+import java.util.logging.Logger;
+
+public class AppAutomationReportBuildAction extends AbstractAppAutomationReportBuildAction {
+    private final static Logger logger = Logger.getLogger(AppAutomationReportBuildAction.class.getName());
+    private String lambdaTestBuildDeviceUrl;
+    private String authString;
+    private String buildName;
+    List<JSONObject> result = new ArrayList<JSONObject>();
+
+    public AppAutomationReportBuildAction(final Run<?, ?> build, String name,String password, String buildName) {
+        super();
+        setBuild(build);
+        this.authString =  name + ":" + password;
+        this.buildName = buildName;
+    }
+    public void generateLambdaTestAppAutomationReport() {
+        logger.info("authString : " + authString);
+        byte[] authEncBytes = Base64.getEncoder().encode(authString.getBytes());
+        String authStringEnc = new String(authEncBytes);
+        logger.info("in generate generateLambdaTestAppAutomationReport function");
+        try {
+            URL buildUrl = new URL(Constant.AppAutomationReport.BUILD_INFO_URL);
+            URLConnection buildUrlConnection = buildUrl.openConnection();
+            buildUrlConnection.setRequestProperty("Authorization", "Basic " + authStringEnc);
+            InputStream is = buildUrlConnection.getInputStream();
+            InputStreamReader isr = new InputStreamReader(is);
+            int numCharsReadForBuild;
+            char[] charArrayForBuild = new char[1024];
+            StringBuffer sb = new StringBuffer();
+            while ((numCharsReadForBuild = isr.read(charArrayForBuild)) > 0) {
+                sb.append(charArrayForBuild, 0, numCharsReadForBuild);
+            }
+            String buildResult = sb.toString();
+            try {
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode rootNode = mapper.readTree(buildResult);
+                ArrayNode dataNode = (ArrayNode) rootNode.get("data");
+                Iterator<JsonNode> dataIterator = dataNode.elements();
+                while (dataIterator.hasNext()) {
+                    JsonNode buildDetailNode = dataIterator.next();
+                    if (buildDetailNode.get("name").toString().replaceAll("\"", "").equals(buildName)) {
+                        String build_id = buildDetailNode.get("build_id").toString();
+
+                        lambdaTestBuildDeviceUrl = "https://appautomation.lambdatest.com/test?buildid=" + build_id;
```

Comment:
Please use constant value `APP_AUTOMATION_APP_URL` here to make URL

Reply:
Done